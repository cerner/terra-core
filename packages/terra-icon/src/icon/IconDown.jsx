/* eslint-disable */
import React from 'react';
import IconBase from '../IconBase';

const SvgIcon = (customProps) => {
  const attributes = Object.assign({}, customProps);

  return (
    <IconBase {...attributes}>
      <path d="M47.9 24.5l-3.8-3.8L27 37.4V0h-6v37.4L3.9 20.7.1 24.5 24 48z" ></path>
    </IconBase>
  );
};

SvgIcon.displayName = "IconDown";
SvgIcon.defaultProps = {"viewBox":"0 0 48 48","xmlns":"http://www.w3.org/2000/svg","data-name":"Layer 1"};

export default SvgIcon;
/* eslint-enable */
