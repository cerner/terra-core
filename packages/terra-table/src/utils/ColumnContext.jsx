import React from 'react';

const ColumnContext = React.createContext({ pinnedColumnOffsets: [] });

export default ColumnContext;
