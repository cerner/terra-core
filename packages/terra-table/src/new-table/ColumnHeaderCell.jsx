import React, { useContext, useRef, useCallback } from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import classNames from 'classnames/bind';

import ThemeContext from 'terra-theme-context';
import IconUp from 'terra-icon/lib/icon/IconUp';
import IconDown from 'terra-icon/lib/icon/IconDown';
import IconError from 'terra-icon/lib/icon/IconError';

import ColumnResizeHandle from './ColumnResizeHandle';
import { SortIndicators } from './proptypes/columnShape';
import ColumnContext from './utils/ColumnContext';
import styles from './ColumnHeaderCell.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Required string representing a unique identifier for the column header cell.
   */
  id: PropTypes.string.isRequired,

  /**
   * String of text to render within the column header cell.
   */
  displayName: PropTypes.string,

  /**
   * A string indicating which sorting indicator should be rendered. If not provided, no sorting indicator will be rendered.
   * If a `component` value is specified, `sortIndicator` will be ignored. One of `ascending`, `descending`.
   */
  sortIndicator: PropTypes.oneOf(Object.values(SortIndicators)),

  /**
   * Boolean value indicating whether or not the column has an error in the data.
   */
  hasError: PropTypes.bool,

  /**
   * Number that specifies the minimum column width in pixels.
   */
  minimumWidth: PropTypes.number,

  /**
   * Number that specifies the maximum column width in pixels.
   */
  maximumWidth: PropTypes.number,

  /**
   * Boolean value indicating whether or not the column header is resizable.
   */
  isResizable: PropTypes.bool,

  /**
   * Height of the parent table.
   */
  tableHeight: PropTypes.number,

  /**
   * String that specifies the default width for columns in the grid. Any valid CSS width value is accepted.
   */
  width: PropTypes.number.isRequired,

  /**
   * String that specifies the column height. Any valid CSS height value accepted.
  */
  headerHeight: PropTypes.string.isRequired,

  /**
   * The cell's row position in the grid. This is zero based.
   */
  rowIndex: PropTypes.number,

  /**
   * The cell's column position in the grid. This is zero based.
   */
  columnIndex: PropTypes.number,

  /**
   * Function that is called when the mouse down event is triggered on the column resize handle.
   */
  onResizeMouseDown: PropTypes.func,

  /**
   * @private
   * Object containing intl APIs
   */
  intl: PropTypes.shape({ formatMessage: PropTypes.func }),
};

const defaultProps = {
  hasError: false,
  isResizable: true,
};

const ColumnHeaderCell = (props) => {
  const {
    id,
    displayName,
    sortIndicator,
    hasError,
    isResizable,
    tableHeight,
    width,
    minimumWidth,
    maximumWidth,
    headerHeight,
    intl,
    rowIndex,
    columnIndex,
    onResizeMouseDown,
  } = props;

  const columnContext = useContext(ColumnContext);
  const columnHeaderCell = useRef();

  const columnHeaderCellRef = useCallback((node) => {
    columnHeaderCell.current = node;
  }, []);

  const onResizeHandleMouseDown = useCallback((event) => {
    if (onResizeMouseDown) {
      onResizeMouseDown(event, columnIndex, columnHeaderCell.current.offsetWidth);
    }
  }, [columnIndex, onResizeMouseDown]);

  let sortIndicatorIcon;
  const errorIcon = hasError && <IconError a11yLabel={intl.formatMessage({ id: 'Terra.dataGrid.columnError' })} className={cx('error-icon')} />;

  // Add the sort indicator based on the sort direction
  if (sortIndicator === SortIndicators.ASCENDING) {
    sortIndicatorIcon = <IconUp />;
  } else if (sortIndicator === SortIndicators.DESCENDING) {
    sortIndicatorIcon = <IconDown />;
  }

  // Retrieve current theme from context
  const theme = useContext(ThemeContext);

  const cellLeftEdge = (columnIndex < columnContext.pinnedColumnOffsets.length) ? columnContext.pinnedColumnOffsets[columnIndex] : null;
  const dividerLeftEdge = width - 1;

  const pinnedColumnsDivider = columnIndex === columnContext.pinnedColumnOffsets.length - 1
    ? (
      <div
        className={cx('pinned-columns-divider')}
        style={{ height: tableHeight, left: dividerLeftEdge }} // eslint-disable-line react/forbid-dom-props
      />
    )
    : null;

  return (
  /* eslint-disable react/forbid-dom-props */
    <th
      ref={(columnHeaderCellRef)}
      key={id}
      className={cx('column-header', theme.className, { pinned: columnIndex < columnContext.pinnedColumnOffsets.length })}
      tabIndex={-1}
      role="columnheader"
      scope="col"
      aria-sort={sortIndicator}
      // eslint-disable-next-line react/forbid-dom-props
      style={{ width: `${width}px`, height: headerHeight, left: cellLeftEdge }}
    >
      <div className={cx('header-container')} role={displayName && 'button'}>
        {errorIcon}
        <span>{displayName}</span>
        {sortIndicatorIcon}
      </div>
      { isResizable && (
      <ColumnResizeHandle
        columnIndex={columnIndex}
        columnText={displayName}
        columnWidth={width}
        height={tableHeight}
        minimumWidth={minimumWidth}
        maximumWidth={maximumWidth}
        onResizeMouseDown={onResizeHandleMouseDown}
      />
      )}
      {pinnedColumnsDivider}
    </th>
  );
};

ColumnHeaderCell.propTypes = propTypes;
ColumnHeaderCell.defaultProps = defaultProps;
export default React.memo(injectIntl(ColumnHeaderCell));
