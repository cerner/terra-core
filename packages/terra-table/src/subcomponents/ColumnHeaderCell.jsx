import React, { useContext, useRef, useCallback } from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import * as KeyCode from 'keycode-js';
import classNames from 'classnames/bind';

import ThemeContext from 'terra-theme-context';
import IconUp from 'terra-icon/lib/icon/IconUp';
import IconDown from 'terra-icon/lib/icon/IconDown';
import IconError from 'terra-icon/lib/icon/IconError';

import ColumnResizeHandle from './ColumnResizeHandle';
import { SortIndicators } from '../proptypes/columnShape';
import ColumnContext from '../utils/ColumnContext';
import styles from './ColumnHeaderCell.module.scss';
import GridContext from '../utils/GridContext';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Required string representing a unique identifier for the column header cell.
   */
  id: PropTypes.string.isRequired,

  /**
   * String of text to render within the column header cell.
   */
  displayName: PropTypes.string,

  /**
   * A string indicating which sorting indicator should be rendered. If not provided, no sorting indicator will be rendered.
   * If a `component` value is specified, `sortIndicator` will be ignored. One of `ascending`, `descending`.
   */
  sortIndicator: PropTypes.oneOf(Object.values(SortIndicators)),

  /**
   * Boolean value indicating whether or not the column has an error in the data.
   */
  hasError: PropTypes.bool,

  /**
   * Number that specifies the minimum column width in pixels.
   */
  minimumWidth: PropTypes.number,

  /**
   * Number that specifies the maximum column width in pixels.
   */
  maximumWidth: PropTypes.number,

  /**
   * Boolean value indicating whether or not the column header is selectable.
  */
  isSelectable: PropTypes.bool,

  /**
   * Boolean value indicating whether or not the column header is resizable.
   */
  isResizable: PropTypes.bool,

  /**
   * Height of the parent table.
   */
  tableHeight: PropTypes.number,

  /**
   * String that specifies the default width for columns in the table. Any valid CSS width value is accepted.
   */
  width: PropTypes.number.isRequired,

  /**
   * String that specifies the column height. Any valid CSS height value accepted.
  */
  headerHeight: PropTypes.string.isRequired,

  /**
   * The cell's row position in the table. This is zero based.
   */
  rowIndex: PropTypes.number,

  /**
   * The cell's column position in the table. This is zero based.
   */
  columnIndex: PropTypes.number,

  /**
   * Function that is called when a selectable header cell is selected. Parameters:
   * @param {string} rowId rowId
   * @param {string} columnId columnId
   */
  onColumnSelect: PropTypes.func,

  /**
   * Function that is called when the mouse down event is triggered on the column resize handle.
   */
  onResizeMouseDown: PropTypes.func,

  /**
   * @private
   * Object containing intl APIs
   */
  intl: PropTypes.shape({ formatMessage: PropTypes.func }),
};

const defaultProps = {
  hasError: false,
  isResizable: true,
  isSelectable: false,
};

const ColumnHeaderCell = (props) => {
  const {
    id,
    displayName,
    sortIndicator,
    hasError,
    isResizable,
    isSelectable,
    tableHeight,
    width,
    minimumWidth,
    maximumWidth,
    headerHeight,
    intl,
    columnIndex,
    onColumnSelect,
    onResizeMouseDown,
    rowIndex,
  } = props;

  const columnContext = useContext(ColumnContext);
  const theme = useContext(ThemeContext);
  const gridContext = useContext(GridContext);
  const columnHeaderCell = useRef();

  const onResizeHandleMouseDown = useCallback((event) => {
    if (onResizeMouseDown) {
      onResizeMouseDown(event, columnIndex, columnHeaderCell.current.offsetWidth);
    }
  }, [columnIndex, onResizeMouseDown]);

  const handleMouseDown = (event) => {
    onColumnSelect(id, { row: rowIndex, col: columnIndex });
    event.stopPropagation();
  };

  const handleKeyDown = (event) => {
    const key = event.keyCode;
    switch (key) {
      case KeyCode.KEY_SPACE:
      case KeyCode.KEY_RETURN:
        onColumnSelect(id, { row: rowIndex, col: columnIndex });
        event.stopPropagation();
        event.preventDefault(); // prevent the default scrolling
        break;
      default:
    }
  };

  const errorIcon = hasError && <IconError a11yLabel={intl.formatMessage({ id: 'Terra.table.columnError' })} className={cx('error-icon')} />;

  let sortIndicatorIcon;
  if (sortIndicator === SortIndicators.ASCENDING) {
    sortIndicatorIcon = <IconUp />;
  } else if (sortIndicator === SortIndicators.DESCENDING) {
    sortIndicatorIcon = <IconDown />;
  }

  const cellLeftEdge = (columnIndex < columnContext.pinnedColumnOffsets.length) ? columnContext.pinnedColumnOffsets[columnIndex] : null;
  const dividerLeftEdge = width - 1;

  const pinnedColumnsDivider = columnIndex === columnContext.pinnedColumnOffsets.length - 1
    ? (
      <div
        className={cx('pinned-columns-divider')}
        style={{ height: tableHeight, left: dividerLeftEdge }} // eslint-disable-line react/forbid-dom-props
      />
    )
    : null;

  return (
  /* eslint-disable react/forbid-dom-props */
    <th
      ref={columnHeaderCell}
      key={id}
      className={cx('column-header', theme.className, { selectable: isSelectable, pinned: columnIndex < columnContext.pinnedColumnOffsets.length })}
      role="columnheader"
      scope="col"
      tabIndex={gridContext.role === 'grid' || isSelectable ? 0 : undefined}
      aria-sort={sortIndicator}
      onMouseDown={(isSelectable && onColumnSelect) ? handleMouseDown : undefined}
      onKeyDown={isSelectable && onColumnSelect ? handleKeyDown : undefined}
      // eslint-disable-next-line react/forbid-dom-props
      style={{ width: `${width}px`, height: headerHeight, left: cellLeftEdge }}
    >
      <div className={cx('header-container')} role={displayName && 'button'}>
        {errorIcon}
        <span>{displayName}</span>
        {sortIndicatorIcon}
      </div>
      { isResizable && (
      <ColumnResizeHandle
        columnIndex={columnIndex}
        columnText={displayName}
        columnWidth={width}
        height={tableHeight}
        minimumWidth={minimumWidth}
        maximumWidth={maximumWidth}
        onResizeMouseDown={onResizeHandleMouseDown}
      />
      )}
      {pinnedColumnsDivider}
    </th>
  );
};

ColumnHeaderCell.propTypes = propTypes;
ColumnHeaderCell.defaultProps = defaultProps;
export default React.memo(injectIntl(ColumnHeaderCell));
