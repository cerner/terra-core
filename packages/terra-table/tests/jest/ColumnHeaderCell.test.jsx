import React from 'react';
import IconUp from 'terra-icon/lib/icon/IconUp';
import IconDown from 'terra-icon/lib/icon/IconDown';
import IconError from 'terra-icon/lib/icon/IconError';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { mountWithIntl, shallowWithIntl } from 'terra-enzyme-intl';
import { IntlProvider } from 'react-intl';
import ColumnHeaderCell from '../../src/subcomponents/ColumnHeaderCell';
import ColumnContext from '../../src/utils/ColumnContext';

describe('ColumnHeaderCell', () => {
  it('renders a default column header cell', () => {
    const column = {
      id: 'Column-0',
      displayName: ' Vitals',
    };

    const wrapper = shallowWithIntl(
      <IntlProvider locale="en">
        <ColumnHeaderCell
          columnIndex={0}
          width={100}
          headerHeight="150px"
          {...column}
        />
      </IntlProvider>,
    ).dive().dive();

    const columnHeader = wrapper.find('.column-header');
    expect(columnHeader).toHaveLength(1);
    expect(columnHeader.key()).toBe('Column-0');
    expect(columnHeader.props().role).toBe('columnheader');
    expect(columnHeader.props().scope).toBe('col');
    expect(columnHeader.props().tabIndex).toEqual(undefined);
    expect(columnHeader.props().style.width).toBe('100px');
    expect(columnHeader.props().style.height).toBe('150px');

    const headerContainer = columnHeader.find('.header-container[role="button"]');
    expect(headerContainer.children()).toHaveLength(1);
    expect(headerContainer.find('span').text().trim()).toBe('Vitals');

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a column header cell with ascending sort', () => {
    const column = {
      id: 'Column-0',
      displayName: ' Vitals',
      sortIndicator: 'ascending',
    };

    const wrapper = shallowWithIntl(
      <IntlProvider locale="en">
        <ColumnHeaderCell
          columnIndex={0}
          width={100}
          headerHeight="150px"
          {...column}
        />
      </IntlProvider>,
    ).dive().dive();

    const columnHeader = wrapper.find('.column-header');
    expect(columnHeader).toHaveLength(1);
    expect(columnHeader.key()).toBe('Column-0');
    expect(columnHeader.props().role).toBe('columnheader');
    expect(columnHeader.props().scope).toBe('col');
    expect(columnHeader.props().tabIndex).toEqual(undefined);
    expect(columnHeader.props()['aria-sort']).toBe('ascending');
    expect(columnHeader.props().style.width).toBe('100px');
    expect(columnHeader.props().style.height).toBe('150px');

    const headerContainer = columnHeader.find('.header-container[role="button"]');
    expect(headerContainer.find('span').text().trim()).toBe('Vitals');
    expect(headerContainer.find(IconUp)).toHaveLength(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a column header cell with descending sort', () => {
    const column = {
      id: 'Column-0',
      displayName: ' Vitals',
      sortIndicator: 'descending',
    };

    const wrapper = shallowWithIntl(
      <IntlProvider locale="en">
        <ColumnHeaderCell
          columnIndex={0}
          width={100}
          headerHeight="150px"
          {...column}
        />
      </IntlProvider>,
    ).dive().dive();

    const columnHeader = wrapper.find('.column-header');
    expect(columnHeader).toHaveLength(1);
    expect(columnHeader.key()).toBe('Column-0');
    expect(columnHeader.props().role).toBe('columnheader');
    expect(columnHeader.props().scope).toBe('col');
    expect(columnHeader.props().tabIndex).toEqual(undefined);
    expect(columnHeader.props()['aria-sort']).toBe('descending');
    expect(columnHeader.props().style.width).toBe('100px');
    expect(columnHeader.props().style.height).toBe('150px');

    const headerContainer = columnHeader.find('.header-container[role="button"]');
    expect(headerContainer.find('span').text().trim()).toBe('Vitals');
    expect(headerContainer.find(IconDown)).toHaveLength(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a column header cell with error', () => {
    const column = {
      id: 'Column-0',
      displayName: ' Vitals',
      hasError: true,
    };

    const wrapper = shallowWithIntl(
      <IntlProvider locale="en">
        <ColumnHeaderCell
          columnIndex={0}
          width={100}
          headerHeight="150px"
          {...column}
        />
      </IntlProvider>,
    ).dive().dive();

    const columnHeader = wrapper.find('.column-header');
    expect(columnHeader).toHaveLength(1);
    expect(columnHeader.key()).toBe('Column-0');
    expect(columnHeader.props().role).toBe('columnheader');
    expect(columnHeader.props().scope).toBe('col');
    expect(columnHeader.props().tabIndex).toEqual(undefined);
    expect(columnHeader.props()['aria-sort']).toBeUndefined();
    expect(columnHeader.props().style.width).toBe('100px');
    expect(columnHeader.props().style.height).toBe('150px');

    const headerContainer = columnHeader.find('.header-container[role="button"]');
    expect(headerContainer.find('span').text().trim()).toBe('Vitals');
    expect(headerContainer.find(IconError)).toHaveLength(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a column header cell with ascending sort and error', () => {
    const column = {
      id: 'Column-0',
      displayName: ' Vitals',
      sortIndicator: 'ascending',
      hasError: true,
    };

    const wrapper = shallowWithIntl(
      <IntlProvider locale="en">
        <ColumnHeaderCell
          columnIndex={0}
          width={100}
          headerHeight="150px"
          {...column}
        />
      </IntlProvider>,
    ).dive().dive();

    const columnHeader = wrapper.find('.column-header');
    expect(columnHeader).toHaveLength(1);
    expect(columnHeader.key()).toBe('Column-0');
    expect(columnHeader.props().role).toBe('columnheader');
    expect(columnHeader.props().scope).toBe('col');
    expect(columnHeader.props().tabIndex).toEqual(undefined);
    expect(columnHeader.props()['aria-sort']).toBe('ascending');
    expect(columnHeader.props().style.width).toBe('100px');
    expect(columnHeader.props().style.height).toBe('150px');

    const headerContainer = columnHeader.find('.header-container[role="button"]');
    expect(headerContainer.find('span').text().trim()).toBe('Vitals');
    expect(headerContainer.find(IconUp)).toHaveLength(1);
    expect(headerContainer.find(IconError)).toHaveLength(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a column header cell with onColumnSelect callback', () => {
    const column = {
      id: 'Column-0',
      displayName: ' Vitals',
      sortIndicator: 'ascending',
      hasError: true,
    };

    const mockClick = jest.fn();

    const wrapper = shallowWithIntl(
      <IntlProvider locale="en">
        <ColumnHeaderCell
          columnIndex={0}
          isSelectable
          width={100}
          headerHeight="150px"
          {...column}
          onColumnSelect={mockClick}
        />
      </IntlProvider>,
    ).dive().dive();

    const columnHeader = wrapper.find('.column-header.selectable');
    expect(columnHeader).toHaveLength(1);
    expect(columnHeader.key()).toBe('Column-0');
    expect(columnHeader.props().role).toBe('columnheader');
    expect(columnHeader.props().scope).toBe('col');
    expect(columnHeader.props().tabIndex).toEqual(0);
    expect(columnHeader.props()['aria-sort']).toBe('ascending');
    expect(columnHeader.props().onMouseDown).toBeDefined();
    expect(columnHeader.props().style.width).toBe('100px');
    expect(columnHeader.props().style.height).toBe('150px');

    const headerContainer = columnHeader.find('.header-container[role="button"]');
    expect(headerContainer.find('span').text().trim()).toBe('Vitals');
    expect(headerContainer.find(IconUp)).toHaveLength(1);
    expect(headerContainer.find(IconError)).toHaveLength(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a column header cell with onColumnSelect callback but not selectable', () => {
    const column = {
      id: 'Column-0',
      displayName: ' Vitals',
      sortIndicator: 'ascending',
      hasError: true,
      isSelectable: false,
    };

    const onColumnSelect = () => {};

    const wrapper = shallowWithIntl(
      <IntlProvider locale="en">
        <ColumnHeaderCell
          columnIndex={0}
          width={100}
          headerHeight="150px"
          {...column}
          onColumnSelect={onColumnSelect}
        />
      </IntlProvider>,
    ).dive().dive();

    const columnHeader = wrapper.find('.column-header:not(selectable)');
    expect(columnHeader).toHaveLength(1);
    expect(columnHeader.key()).toBe('Column-0');
    expect(columnHeader.props().role).toBe('columnheader');
    expect(columnHeader.props().scope).toBe('col');
    expect(columnHeader.props().tabIndex).toEqual(undefined);
    expect(columnHeader.props()['aria-sort']).toBe('ascending');
    expect(columnHeader.props().onMouseDown).toBeUndefined();
    expect(columnHeader.props().style.width).toBe('100px');
    expect(columnHeader.props().style.height).toBe('150px');

    const headerContainer = columnHeader.find('.header-container[role="button"]');
    expect(headerContainer.find('span').text().trim()).toBe('Vitals');
    expect(headerContainer.find(IconUp)).toHaveLength(1);
    expect(headerContainer.find(IconError)).toHaveLength(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a pinned column header cell', () => {
    jest.spyOn(console, 'error').mockImplementation(); // eslint-disable-line no-console

    const column = {
      id: 'Column-0',
      displayName: ' Vitals',
      sortIndicator: 'ascending',
      hasError: true,
    };

    const wrapper = mountWithIntl(
      <ColumnContext.Provider value={{ pinnedColumnOffsets: [0] }}>
        <ColumnHeaderCell
          columnIndex={0}
          width={100}
          headerHeight="150px"
          {...column}
        />
      </ColumnContext.Provider>,
    );

    expect(wrapper.find('.pinned')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();

    console.error.mockRestore(); // eslint-disable-line no-console
  });
});
