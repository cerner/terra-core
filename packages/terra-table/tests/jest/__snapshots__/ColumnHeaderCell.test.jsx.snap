// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnHeaderCell renders a column header cell with ascending sort 1`] = `
<th
  aria-sort="ascending"
  className="column-header selectable"
  key="Column-0"
  role="columnheader"
  scope="col"
  style={
    Object {
      "height": "150px",
      "left": null,
      "width": "100px",
    }
  }
  tabIndex={0}
>
  <div
    className="header-container"
    role="button"
  >
    <span>
       Vitals
    </span>
    <IconUp
      data-name="Layer 1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
  <InjectIntl(ColumnResizeHandle)
    columnIndex={0}
    columnText=" Vitals"
    columnWidth={100}
    onResizeMouseDown={[Function]}
  />
</th>
`;

exports[`ColumnHeaderCell renders a column header cell with ascending sort and error 1`] = `
<th
  aria-sort="ascending"
  className="column-header"
  key="Column-0"
  role="columnheader"
  scope="col"
  style={
    Object {
      "height": "150px",
      "left": null,
      "width": "100px",
    }
  }
>
  <div
    className="header-container"
    role="button"
  >
    <IconError
      a11yLabel="Terra.table.columnError"
      className="error-icon"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    <span>
       Vitals
    </span>
    <IconUp
      data-name="Layer 1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
  <InjectIntl(ColumnResizeHandle)
    columnIndex={0}
    columnText=" Vitals"
    columnWidth={100}
    onResizeMouseDown={[Function]}
  />
</th>
`;

exports[`ColumnHeaderCell renders a column header cell with descending sort 1`] = `
<th
  aria-sort="descending"
  className="column-header selectable"
  key="Column-0"
  role="columnheader"
  scope="col"
  style={
    Object {
      "height": "150px",
      "left": null,
      "width": "100px",
    }
  }
  tabIndex={0}
>
  <div
    className="header-container"
    role="button"
  >
    <span>
       Vitals
    </span>
    <IconDown
      data-name="Layer 1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
  <InjectIntl(ColumnResizeHandle)
    columnIndex={0}
    columnText=" Vitals"
    columnWidth={100}
    onResizeMouseDown={[Function]}
  />
</th>
`;

exports[`ColumnHeaderCell renders a column header cell with error 1`] = `
<th
  className="column-header"
  key="Column-0"
  role="columnheader"
  scope="col"
  style={
    Object {
      "height": "150px",
      "left": null,
      "width": "100px",
    }
  }
>
  <div
    className="header-container"
    role="button"
  >
    <IconError
      a11yLabel="Terra.table.columnError"
      className="error-icon"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    <span>
       Vitals
    </span>
  </div>
  <InjectIntl(ColumnResizeHandle)
    columnIndex={0}
    columnText=" Vitals"
    columnWidth={100}
    onResizeMouseDown={[Function]}
  />
</th>
`;

exports[`ColumnHeaderCell renders a column header cell with onColumnSelect callback 1`] = `
<th
  aria-sort="ascending"
  className="column-header selectable"
  key="Column-0"
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="columnheader"
  scope="col"
  style={
    Object {
      "height": "150px",
      "left": null,
      "width": "100px",
    }
  }
  tabIndex={0}
>
  <div
    className="header-container"
    role="button"
  >
    <IconError
      a11yLabel="Terra.table.columnError"
      className="error-icon"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    <span>
       Vitals
    </span>
    <IconUp
      data-name="Layer 1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
  <InjectIntl(ColumnResizeHandle)
    columnIndex={0}
    columnText=" Vitals"
    columnWidth={100}
    onResizeMouseDown={[Function]}
  />
</th>
`;

exports[`ColumnHeaderCell renders a column header cell with onColumnSelect callback but not selectable 1`] = `
<th
  aria-sort="ascending"
  className="column-header"
  key="Column-0"
  role="columnheader"
  scope="col"
  style={
    Object {
      "height": "150px",
      "left": null,
      "width": "100px",
    }
  }
>
  <div
    className="header-container"
    role="button"
  >
    <IconError
      a11yLabel="Terra.table.columnError"
      className="error-icon"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    <span>
       Vitals
    </span>
    <IconUp
      data-name="Layer 1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
  <InjectIntl(ColumnResizeHandle)
    columnIndex={0}
    columnText=" Vitals"
    columnWidth={100}
    onResizeMouseDown={[Function]}
  />
</th>
`;

exports[`ColumnHeaderCell renders a default column header cell 1`] = `
<th
  className="column-header"
  key="Column-0"
  role="columnheader"
  scope="col"
  style={
    Object {
      "height": "150px",
      "left": null,
      "width": "100px",
    }
  }
>
  <div
    className="header-container"
    role="button"
  >
    <span>
       Vitals
    </span>
  </div>
  <InjectIntl(ColumnResizeHandle)
    columnIndex={0}
    columnText=" Vitals"
    columnWidth={100}
    onResizeMouseDown={[Function]}
  />
</th>
`;

exports[`ColumnHeaderCell renders a pinned column header cell 1`] = `
<InjectIntl(ColumnHeaderCell)
  columnIndex={0}
  displayName=" Vitals"
  hasError={true}
  headerHeight="150px"
  id="Column-0"
  sortIndicator="ascending"
  width={100}
>
  <ColumnHeaderCell
    columnIndex={0}
    displayName=" Vitals"
    hasError={true}
    headerHeight="150px"
    id="Column-0"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    isResizable={true}
    isSelectable={false}
    sortIndicator="ascending"
    width={100}
  >
    <th
      aria-sort="ascending"
      className="column-header pinned"
      key="Column-0"
      role="columnheader"
      scope="col"
      style={
        Object {
          "height": "150px",
          "left": 0,
          "width": "100px",
        }
      }
    >
      <div
        className="header-container"
        role="button"
      >
        <IconError
          a11yLabel="Terra.table.columnError"
          className="error-icon"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <IconBase
            a11yLabel="Terra.table.columnError"
            className="IconError error-icon"
            focusable="false"
            height="1em"
            isBidi={false}
            isSpin={false}
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              className="tui-Icon icon IconError error-icon"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Terra.table.columnError
              </title>
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </IconBase>
        </IconError>
        <span>
           Vitals
        </span>
        <IconUp
          data-name="Layer 1"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <IconBase
            data-name="Layer 1"
            focusable="false"
            height="1em"
            isBidi={false}
            isSpin={false}
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              className="tui-Icon icon"
              data-name="Layer 1"
              focusable="false"
              height="1em"
              role="presentation"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0 .1 23.5l3.8 3.8L21 10.6V48h6V10.6l17.1 16.7 3.8-3.8z"
              />
            </svg>
          </IconBase>
        </IconUp>
      </div>
      <InjectIntl(ColumnResizeHandle)
        columnIndex={0}
        columnText=" Vitals"
        columnWidth={100}
        onResizeMouseDown={[Function]}
      >
        <ColumnResizeHandle
          columnIndex={0}
          columnText=" Vitals"
          columnWidth={100}
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": null,
              "now": [Function],
              "onError": [Function],
              "textComponent": "span",
              "timeZone": null,
            }
          }
          onResizeMouseDown={[Function]}
        >
          <div
            aria-label=" Vitals"
            aria-valuenow={100}
            aria-valuetext="Terra.table.resizeHandleValueText"
            className="resize-handle"
            draggable={true}
            onMouseDown={[Function]}
            role="slider"
            style={
              Object {
                "height": "undefinedpx",
              }
            }
            tabIndex={0}
          />
        </ColumnResizeHandle>
      </InjectIntl(ColumnResizeHandle)>
      <div
        className="pinned-columns-divider"
        style={
          Object {
            "height": undefined,
            "left": 99,
          }
        }
      />
    </th>
  </ColumnHeaderCell>
</InjectIntl(ColumnHeaderCell)>
`;
